### PR TITLE
Pybuffer

### PR DESCRIPTION
**Pull request by koubaa**
### What does this implement/fix? Explain your changes.
Rebase of https://github.com/pythonnet/pythonnet/pull/980/files 
...

### Does this close any currently open issues?

...

### Any other comments?

...

### Checklist

Check all those that are applicable and complete.

-   [ ] Make sure to include one or more tests for your change
-   [ ] If an enhancement PR, please create docs and at best an example
-   [ ] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [ ] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by koubaa** [comment]
@filmor I took this one

File: src/runtime/runtime.cs [hunk 1]
```diff
@@ -147,6 +147,19 @@ public class Runtime
...
 
+        internal static int pyversionnumber
+        {
+            get
+            {
+                var versionString = PythonEngine.Version;
+                //"3.0a5+ (py3k:63103M, May 12 2008, 00:53:55) \n[GCC 4.2.3]"
+
+                //we only support python 3 so we just need to check the third character
+                return 30 + Convert.ToInt32(versionString[2]);
```
Comment (filmor) [inline src/runtime/runtime.cs:158]: This seems very brittle to me, there must be a better solution. Also, we don't support any Python version before 3.5, so checks for that can go.

Comment (koubaa) [reply src/runtime/runtime.cs:158]: @filmor I could use sys.version_info but I'd rather not run python code.  Let me look for a C API equivalent

Comment (koubaa) [reply src/runtime/runtime.cs]: @filmor see now

File: src/runtime/runtime.cs [hunk 2]
```diff
@@ -147,6 +147,18 @@ public class Runtime
 
         private static PyReferenceCollection _pyRefs = new PyReferenceCollection();
 
+        internal static int pyversionnumber
```
Comment (lostmsu) [inline src/runtime/runtime.cs:150]: Can you use `System.Version` here? `major * 10 + minor` led to compatibility issue in many products before. This is not going to work for 3.10

Comment (koubaa) [reply src/runtime/runtime.cs]: @lostmsu great point.  Updated

**Comment by koubaa** [comment]
@lostmsu please review

File: src/runtime/pybuffer.cs [hunk 3]
```diff
@@ -88,7 +88,7 @@ private static char OrderStyleToChar(BufferOrderStyle order, bool eitherOneValid
         /// </summary>
         public static long SizeFromFormat(string format)
         {
-            if (Runtime.pyversionnumber < 39)
+            if (Runtime.PyVersion.Minor < 9)
```
Comment (lostmsu) [inline src/runtime/pybuffer.cs:91]: You must compare the full version. E.g. `Runtime.PyVersion < new Version(3, 9)`

Comment (koubaa) [reply src/runtime/pybuffer.cs]: @lostmsu done

**Pull request merged**